src/raylib/shapes/Circle.hpp

```diff
@@ -21,117 +21,113 @@ namespace raylib
+            /// segments means a more detailed circle.
+            /// @param color The color of the lines
+            void drawSectorLines(float startAngle, float endAngle, float segments, Color color);
+
+            /// Draw a circle with a gradient
+            ///
+            /// @param color1 The color of the circle's center
+            /// @param color2 The color of the outer circle
+            void drawGradient(Color color1, Color color2);
+
+            /// DrawCircleLines draws a circle with the given parameters
+            ///
+            /// @param color Color of the circle to draw (it can be a Color struct or a
+            /// number in hexadecimal format 0xRRGGBB)
+            void drawLines(Color color);
+
+            /// > Check if two circles are colliding
+            ///
+            /// @param other The other circle to check for collision with.
+            bool checkCollisions(Circle *other);
+
+            /// If the distance between the centers of the two circles is less than the
+            /// sum of their radii, the circles are colliding
+            ///
+            /// @param other The other rectangle to check against.
+            bool checkCollisions(Rectangle other);
+
+            /// Draw a circle in 3D space with a given rotation axis and angle
+            ///
+            /// @param rotationAxis The axis of rotation.
+            /// @param rotationAngle The angle of rotation in degrees.
+            void draw3D(Vector3 rotationAxis, float rotationAngle);
+
+            /// This function sets the position of the circle.
+            ///
+            /// @param x The x position of the circle.
+            /// @param y The y position of the circle.
+            /// @param z The z-coordinate of the circle.
+            void setPosition(float x, float y, float z = 0);
+
+            /// `setRadius` is a function that takes a float as an argument and sets the
+            /// value of the private variable `_radius` to the value of the argument
+            ///
+            /// @param radius The radius of the circle.
+            void setRadius(float radius);
+
+            /// This function sets the color of the circle to the color passed in.
+            ///
+            /// @param color The color of the circle.
+            void setColor(Color color);
+
+            /// It returns the position of the circle.
+            ///
+            /// @return A Vector3 object.
+            Vector3 getPosition();
+
+            /// `getRadius()` returns the value of the private variable `_radius`
+            ///
+            /// @return The radius of the circle.
+            float getRadius();
+
+            /// `getColor()` returns the color of the circle
+            ///
+            /// @return The color of the circle.
+            Color getColor();
+
+          private:
+            float _x;
+            float _y;
+            float _z;
```

Comment:
For all positional characteristics: use one Vector3 instead of three separate variables.